stdlib/assoc-seq.mc

```diff
@@ -1,14 +1,42 @@
 -- A simple library of functions operating on associative sequences. The
 -- difference compared to assoc.mc is that the data type contained here is
--- ordered. With more recently inserted bidings shadowing previous bindings,
+-- ordered. With more recently inserted bindings shadowing previous bindings,
 -- insertion is O(1). This makes this data type suitable for e.g., evaluation
 -- environments and similar.
 
+include "seq.mc"
+
 type AssocSeq k v = [(k, v)]
 type AssocTraits k = {eq: k -> k -> Bool}
 
-let assocSeqInsert: k -> v -> AssocSeq k v -> AssocSeq k v =
+let assocSeqEmpty : AssocSeq k v = []
```

Comment:
I'm not sure if we need to have this layer of abstraction for everything here (as we did in assocMap). My idea was that this file should simply contain a set of convenience functions over lists of tuples (as OCaml does with the `assoc` and `mem_assoc` functions.

In summary: I propose we simply use `[]` directly without having an `assocSeqEmpty`.

Reply:
Yeah, that sounds better. I pushed a commit where the functions you mentioned were removed.